packages/react-composites/src/composites/CallComposite/CallComposite.tsx

```diff
@@ -64,6 +64,30 @@ export interface CallCompositeProps extends BaseCompositeProps<CallCompositeIcon
   role?: Role;
 }
 
+/* @conditional-compile-remove(call-readiness) */
+/**
+ * Device Permission restrictions.
+ * Be able to start a call depending on camera and microphone permission options.
+ *
+ * @beta
+ */
+export interface DevicePermissionRestrictions {
+  /**
+   * Camera Permission prompts for your call.
+   * 'required' - requires the permission to be allowed before permitting the user join the call.
+   * 'optional' - permission can be disallowed and the user is still permitted to join the call.
+   * 'doNotPrompt' - permission is not required and the user is not prompted to allow the permission.
+   */
+  camera: 'required' | 'optional' | 'doNotPrompt';
+  /**
+   * Microphone permission prompts for your call.
+   * 'required' - requires the permission to be allowed before permitting the user join the call.
+   * 'optional' - permission can be disallowed and the user is still permitted to join the call.
+   * 'doNotPrompt' - permission is not required and the user is not prompted to allow the permission.
+   */
+  microphone: 'required' | 'optional' | 'doNotPrompt';
```

Comment:
I guess remove this based on calling SDK limitation?

Reply:
Technically this microphone device permission still "works" with "askDevicePermission".
askDevicePermission asks for camera and microphone independently of what the call type actually requires.
So you can essentially prompt for devices in any page outside of a call using askDevicePermission. But if microphone was not "allowed" then when starting/joining a call will prompt for Microphone regardless.